Whiskey/Tasks/Set-WhiskeyBitbucketBuildStatus.ps1

```diff
@@ -0,0 +1,65 @@
+
+function Set-WhiskeyBitbucketBuildStatus
+{
+    [CmdletBinding()]
+    [Whiskey.Task('PublishBuildStatusToBitbucket')]
+    param(
+        [Parameter(Mandatory=$true)]
+        [Whiskey.Context]
+        $TaskContext,
+
+        [Parameter(Mandatory=$true)]
+        [hashtable]
+        $TaskParameter
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    if( $TaskContext.ByDeveloper )

```

Comment:
Should this be built-in like this or should we let/make the user use the `OnlyBy` global property?

Reply:
I couldn't think of a scenario when a developer would ever want to run a build locally to update the build status of a commit on an SCM server, so I decided this task only makes sense for a build server to prevent any accidental publishing.

Or at least the default should be to not publish build status as a developer and maybe offer an override property?

Reply:
OK. Makes sense. Leave it and make sure it is documented in the task that it will only ever run on a build server.